### PR TITLE
chore(skore)!: Rename `cv_splitter` to `splitter` in cross-validation

### PR DESCRIPTION
**Pull request by thomass-dev**
> [!CAUTION]
> Must be merged after https://github.com/probabl-ai/skore/issues/1955.

> Can i rename `cv_splitter` in `splitter` 👼🏻 ?
> It looks obvious that splitter in the `CrossValidationReport` is the splitter of the cross-validation.

_Originally posted by @thomass-dev in https://github.com/probabl-ai/skore/pull/1956#discussion_r2245737102_

### CODE REVIEW

**Comment by MarieSacksick** [comment]
Waiting for a review from @glemaitre to double-check this API change, but to me it would be fine!  
Historically, in scikit-learn, this parameter is called "cv" (cf function [cross_validate](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html)), hence the reason why we extended the param name, to make it clear but keeps some roots with what people are used to. But seeing your proposition, I find it fine to change.

**Review by glemaitre — APPROVED** [review]
+1 with the change.

I see that it makes more sense that `splitter` is related to the "cross-validation" of the report. I can think of a similar renaming from `scoring` (that means higher is always better) to `metric` (that can be both side). But this is for another PR.
